modules/grpc/otel/otel-protobuf-formatter.cpp

```diff
@@ -584,18 +584,16 @@ ProtobufFormatter::set_syslog_ng_macros(LogMessage *msg, LogRecord &log_record)
   m->set_key("m");
   KeyValueList *macros_kvlist = m->mutable_value()->mutable_kvlist_value();
 
-  LogMessageValueType type;
-  gssize len;
-  const char *value;
-
   KeyValue *pri_attr = macros_kvlist->add_values();
   pri_attr->set_key("PRI");
   pri_attr->mutable_value()->set_int_value(msg->pri);
 
-  value = log_msg_get_value_by_name_with_type(msg, "TAGS", &len, &type);
+  GString *tags_value = g_string_sized_new(64);
```

Comment:
[optional]

I know that gRPC already has a big memory allocation overhead, but I would try to keep the memory allocations minimal where we are able to.

In this case I would either introduce scratch buffers on the OTLP protobuf formatting side, or I would add a dedicated macro for global tags (`$GLOBAL_TAGS` ?) and use `log_msg_get_value_by_name_with_type()` here, and let `LogMessage::log_msg_get_macro_value()`'s `value = g_private_get(&priv_macro_value);` optimization take care of it.

Reply:
I don't want to use the by_name() versions of those functions, that means an extra lookup (name->handle->nvtable)